ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/AvroDiffy.scala

```diff
@@ -49,9 +54,22 @@ class AvroDiffy[T <: GenericRecord](ignore: Set[String] = Set.empty,
             diff(a, b, fullName)
           }
         case Schema.Type.ARRAY if unordered.contains(fullName) =>
-          val a = sortList(x.get(name).asInstanceOf[java.util.List[AnyRef]])
-          val b = sortList(y.get(name).asInstanceOf[java.util.List[AnyRef]])
-          if (a == b) Nil else Seq(Delta(fullName, a, b, delta(a, b)))
+          if (f.schema().getElementType.getType == Schema.Type.RECORD
+              && unordered.exists(_.startsWith(s"$fullName."))
```

Comment:
don't we already check a stricter check above `case Schema.Type.ARRAY if unordered.contains(fullName)` is stricter than `unordered.exists(_.startsWith(s"$fullName."))` so this seems unnecessary then potentially

Reply:
This is to check if there is a nested field that should be unordered, which is the case where we need to properly sort and recurse to compare. If there is nothing with `fullName.` then the assuming is we can sort this regularly since all fields should be precise